source/clear-linux/concepts/restart.rst

```diff
@@ -0,0 +1,184 @@
+
+Restart system services after an OS update
+##########################################
+
+The software life cycle describes how software is created, developed, and
+deployed, and includes how to replace or update software. A good OS
+provides tools for the entire software life cycle. These tools must include
+ways to remove software components properly when replaced with something else.
+
+Most of the work on software update code in |CL| has primarily focused on
+adding new software to the system. We recommend that users reboot their system
+once in a while, but we did not provide any tools to restart services easily,
+until now.
+
+User challenges
+***************
+
+It is difficult to determine which services to restart. You can either
+evaluate each system and reboot manually, or figure out which services to
+restart based on documentation like the |CL| release notes. Since neither
+option solves the issue completely, the |CL| team created a solution.
+
+Over the years, several OSes approached the problem and created partial
+solutions such as the following:
+
+* Automatically restart services during an upgrade.
+* Evaluate services using these steps:
+
+  * Mark updates requiring a reboot, such as kernel updates.
+  * Inform the user of those updates.
+  * Ask the user to restart.
+
+Both solutions are acceptable for many OSes. However, |CL| updates software
+automatically and users do not see notices from the updater unless they review
+the journal.
+
+|CL| requires a completely different solution, with the following
+requirements:
+
+* Eliminate the guesswork about what to restart and under what circumstances.
+* Cannot restart everything. Many service daemons do not support an automatic
+  background restart.
+* Fit into the |CL| architectural perspective: be small, quick, and lean.
+
+:command:`clr-service-restart` functionality
+********************************************
+
+Typical reasons to restart a service daemon include:
+
+* A new version replaces the executable file itself.
+* A new version replaces a library component used by a service daemon.
+
+Our method restarts daemons for both cases by reading various files in the
+:file:`procfs` filesystem provided by the kernel.
+
+The second part of the problem is to determine whether or not running
+processes are part of a system service. The tool focuses on system services
+because most system services are background tasks with no direct user
+interaction. Fortunately, :command:`systemd` provides a simple way to:
+
+* Determine which active tasks are within the system domain.
+* Determine which service daemon maps to the task.
+
+We combined both solutions into a low-overhead tool that shows which system
+daemons require a restart, as shown below:
+
+    Figure 1: Invoke :command:`clr-service-restart`.
+
+    .. code-block:: bash
+
+       sudo clr-service-restart -a -n
+
+    .. code-block:: console
+
+      upower.service: needs a restart (a library dependency was updated)
```

Comment:
Looks like this paragraph needs one more space in front of each line to properly render